trl/trainer/dpo_trainer.py

```diff
@@ -762,22 +771,18 @@ def build_tokenized_answer(self, prompt, answer, images=None):
         answer_input_ids = full_tokenized["input_ids"][response_token_ids_start_idx:]
         answer_attention_mask = full_tokenized["attention_mask"][response_token_ids_start_idx:]
 
+        return_dict = dict(
+            prompt_input_ids=prompt_input_ids,
+            prompt_attention_mask=prompt_attention_mask,
+            input_ids=answer_input_ids,
+            attention_mask=answer_attention_mask,
+        )
         if "pixel_values" in full_tokenized:
-            return dict(
-                prompt_input_ids=prompt_input_ids,
-                prompt_attention_mask=prompt_attention_mask,
-                prompt_pixel_values=full_tokenized["pixel_values"],
-                prompt_pixel_attention_mask=full_tokenized["pixel_attention_mask"],
-                input_ids=answer_input_ids,
-                attention_mask=answer_attention_mask,
-            )
-        else:
-            return dict(
-                prompt_input_ids=prompt_input_ids,
-                prompt_attention_mask=prompt_attention_mask,
-                input_ids=answer_input_ids,
-                attention_mask=answer_attention_mask,
-            )
+            return_dict["prompt_pixel_values"] = full_tokenized["pixel_values"]
+        if "pixel_attention_mask" in full_tokenized:
+            return_dict["prompt_pixel_attention_mask"] = full_tokenized["pixel_attention_mask"]
+
+        return return_dict
```

Comment:
Equivalent, but simpler. The only motif is that it allows `full_tokenized` not to have an entry `"pixel_attention_mask"` (which is the case for Llava)